offchain/authority-claimer/src/listener.rs

```diff
@@ -0,0 +1,62 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+use async_trait::async_trait;
+use rollups_events::{BrokerConfig, DAppMetadata, RollupsClaim};
+use snafu::Snafu;
+use std::fmt::Debug;
+
+use crate::metrics::AuthorityClaimerMetrics;
+
+/// The `BrokerListener` listens for new claims from the broker.
+///
+/// The `listen` function should preferably yield to other processes while
+/// waiting for new messages (instead of busy-waiting).
+#[async_trait]
+pub trait BrokerListener: Sized + Send + Debug {
```

Comment:
Given the broker pattern, we could rename this trait to `ClaimConsumer`, the `listen` function to `consume`, and the struct `DefaultBrokerListener` to `BrokerClaimConsumer`.

Reply:
I chose to use `Broker` because it highlights for people that are not familiar with the architecture that we are communicating with the broker. Using `Claim` is problematic for me because of the directionality of stuff. Reading `ClaimConsumer` I'm left with a micro question inside my head that says "consuming from the blockchain or from the broker?". I know it sounds silly, but it bugs me to have to remember that every time. And if it bugs me, imagine how people that are new to the code will feel.

And `Listener` because it clearly informs that there is some event-like loop shenanigans inside.

There will probably be a `ClaimConsumer` inside the listener when we code it.

Reply:
After writing this, I also think that the name `ClaimSender` is bad. Hehehe.
Let's discuss this today in the meeting.